test/integration/ThreeChainEndToEnd.t.sol

```diff
@@ -0,0 +1,202 @@
+
+}
+
+/// @title  Three Chain End-to-End Test
+/// @notice Extends the dual-chain setup to include a third chain (C) which acts as an additional spoke
+///         Hub is on Chain A, with spokes on Chains B and C
+///         C is considered the source chain, B the destination chain
+///         Depending on the cross chain direction, the hub is either on A or B or C
+contract ThreeChainEndToEndDeployment is EndToEndUtils {
+    using CastLib for *;
+    using MessageLib for *;
+
+    uint16 constant CENTRIFUGE_ID_C = 7;
+    ISafe immutable safeAdminC = ISafe(makeAddr("SafeAdminC"));
+
+    FullDeployer deployC = new FullDeployer();
+    LocalAdapter adapterCToA;
+    LocalAdapter adapterAToC;
+
+    CSpoke sC;
+    CSpoke sB;
+
+    function setUp() public override {
+        // Call the original setUp to set up chains A and B
+        super.setUp();
+
+        // Deploy the third chain (C)
+        adapterCToA = _deployChain(deployC, CENTRIFUGE_ID_C, CENTRIFUGE_ID_A, safeAdminC);
+        _setSpoke(deployC, CENTRIFUGE_ID_C, sC);
+
+        // Connect Chain A to Chain C (spoke 2)
+        adapterAToC = new LocalAdapter(CENTRIFUGE_ID_A, deployA.multiAdapter(), address(deployA));
+        vm.startPrank(address(h.guardian.safe()));
+        deployA.wire(CENTRIFUGE_ID_C, adapterAToC, address(adapterAToC));
+        vm.stopPrank();
+
+        adapterCToA.setEndpoint(adapterAToC);
+        adapterAToC.setEndpoint(adapterCToA);
+
+        vm.label(address(adapterAToC), "AdapterAToC");
+        vm.label(address(adapterCToA), "AdapterCToA");
+    }
+
+    function _setSpokes(CrossChainDirection direction) internal {
+        if (direction == CrossChainDirection.WithIntermediaryHub) {
+            _setSpoke(deployB, CENTRIFUGE_ID_B, sB);
+            _setSpoke(deployC, CENTRIFUGE_ID_C, sC);
+        } else if (direction == CrossChainDirection.FromHub) {
+            _setSpoke(deployB, CENTRIFUGE_ID_B, sB);
+            _setSpoke(deployA, CENTRIFUGE_ID_A, sC);
+        } else if (direction == CrossChainDirection.ToHub) {
+            _setSpoke(deployA, CENTRIFUGE_ID_A, sB);
+            _setSpoke(deployC, CENTRIFUGE_ID_C, sC);
+        }
+    }
+
+    /// @notice Configure the third chain (C) with assets
+    function _testConfigureAssets(CrossChainDirection direction) internal {
+        _setSpokes(direction);
+
+        _configureAsset(sB);
+        _configureAsset(sC);
+    }
+
+    /// @notice Configure a pool with support for all three chains
+    function _testConfigurePool(CrossChainDirection direction) internal {
+        _setSpokes(direction);
+
+        _configurePool(sB);
+        _configurePool(sC);
+    }
+
+    /// @notice Test transferring shares between Chain B and Chain C via Hub A
+    function _testCrossChainTransferShares(CrossChainDirection direction) internal {
+        uint128 amount = 1e18;
+
+        _testConfigurePool(direction);
+
+        // B: Mint shares
+        vm.startPrank(address(sB.root));
```

Comment:
I've used in other sections `vm.startPrank(DEPLOYER);` to indicate that we're somehow adding a line that can not be done in production. I would leave `vm.startPrank(address(sB.root));` for cases where actually we need root because we mimic some real use case in that way. WDYT?